internal/factsengine/gatherers/sbddump.go

```diff
@@ -0,0 +1,50 @@
+package gatherers
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/utils"
+)
+
+const (
+	SBDDumpFactKey = "SBD_dump"
+)
+
+type SBDDumpGatherer struct {
+	executor CommandExecutor
+}
+
+func NewSBDDumpGatherer() *SBDDumpGatherer {
```

Comment:
The constructor function should have the dependency as parameter, in order to inject that. So a parameter of CommandExecutor type is needed. This way the constructor function wires a dependency that cannot be changed so it defeats the purpose of having a dependency injected and declared.

With this change you can inject the mock in tests, and do the test in the separate test package, and initialize the gatherer with the proper function, injecting the mocked dependency instead of initialize the struct without the constructor and use the private scope of the same package :D

Reply:
I'd just go for this one as-is and switch to using DI with CommandExecutor next week on a follow-up PR, wdyt @CDimonaco ?